project-base/src/Shopsys/ShopBundle/Controller/Front/CartController.php

```diff
@@ -103,6 +103,8 @@ public function indexAction(Request $request)
                 }
             } catch (\Shopsys\FrameworkBundle\Model\Cart\Exception\InvalidQuantityException $ex) {
                 $invalidCart = true;
+            } catch (\Shopsys\FrameworkBundle\Model\Cart\Exception\CartIsEmptyException $ex) {
+                $this->getFlashMessageSender()->addErrorFlashTwig(t('We are sorry, products added in cart are not available anymore.'));
```

Comment:
You are adding another flash messaage here (see image bellow). Just catching an exception is enough (with no action). Message `Product 22" Sencor SLE 22F46DM4 HELLO KITTY you had in cart is no longer available. Please check your order.` is added as well.

![Screenshot_2019-11-11 Cart Demo eshop](https://user-images.githubusercontent.com/53976269/68603904-8bd1fb80-04a9-11ea-808e-5059fd17b768.png)

Reply:
Hello, probably you missed translations here.

Reply:
Thx. fixed.